### PR TITLE
fontforge: fix error: Core python package 'pkg_resources' not found: Cannot discover plugins

### PR DESCRIPTION
**Pull request by UlyssesZh**
Tested both `fontforge` and `fontforge-gtk`. No problem anymore. I can still run Python script using Fontforge GUI.

First report of this problem: https://github.com/NixOS/nixpkgs/pull/163439#issuecomment-1062928123. @risicle suggested (#291941) that we fix this.

@erictapen Could you please see whether my fix fits the semantics of the `withPython` argument?

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by philiptaron** [comment]
Needs to target staging due to the number of rebuilds.

File: pkgs/tools/misc/fontforge/default.nix [hunk 1]
```diff
@@ -46,16 +71,40 @@ stdenv.mkDerivation rec {
...
+    cmake
+    makeWrapper
+  ];
+  buildInputs =
+    [
+      readline
+      uthash
+      woff2
+      zeromq
+      py
```
Comment (philiptaron) [inline pkgs/tools/misc/fontforge/default.nix:85]: I'll test this out, but I think this should be in a `withPython` conditional.

File: pkgs/tools/misc/fontforge/default.nix [hunk 2]
```diff
@@ -67,16 +116,20 @@ stdenv.mkDerivation rec {
   '';
 
   postInstall =
-    # get rid of the runtime dependency on python
-    lib.optionalString (!withPython) ''
-      rm -r "$out/share/fontforge/python"
-    '';
+    ''
+      wrapProgram $out/bin/fontforge --suffix PYTHONPATH : "${py}/${py.sitePackages}"
+    ''
```
Comment (philiptaron) [inline pkgs/tools/misc/fontforge/default.nix:121]: And this should be in a withPython conditional too

**Review by philiptaron — CHANGES_REQUESTED** [review]
Requesting changes for moving to the staging branch; let's explore if the Python stuff can be made conditional.

File: pkgs/tools/misc/fontforge/default.nix (hunk 2 above)
Comment (UlyssesZh) [reply pkgs/tools/misc/fontforge/default.nix:121]: I would like to first know what `$out/share/fontforge/python` does. Do you know that?

Comment (philiptaron) [reply pkgs/tools/misc/fontforge/default.nix:121]: No, I haven't jumped into the fontforge deep end yet. 🙂

Comment (UlyssesZh) [reply pkgs/tools/misc/fontforge/default.nix:121]: OK. By the way I shoudn't have replied here. Let's discuss this outside, directly under the PR.

**Review by philiptaron — APPROVED** [review]
I spent a while testing this out. This is a fine way to go until the next release of FontForge, which has more Python modernization.

**Pull request merged**